protocol/src/types/cstring.rs

```diff
@@ -0,0 +1,61 @@
+use std::ffi::CString;
+use std::io::prelude::{Read, Write};
+use {hint, Error, Parcel, Settings};
+
+impl Parcel for CString {
+    const TYPE_NAME: &'static str = "CString";
+
+    fn read_field(
+        read: &mut dyn Read,
+        settings: &Settings,
+        _hints: &mut hint::Hints,
+    ) -> Result<Self, Error> {
+        let mut result = Vec::new();
+        loop {
+            let c: u8 = Parcel::read(read, settings)?;
+            if c == 0x00 {
+                break;
+            }
+            result.push(c);
+        }
+        Ok(CString::new(result)?)
+    }
+
+    fn write_field(
+        &self,
+        write: &mut dyn Write,
+        settings: &Settings,
+        _hints: &mut hint::Hints,
+    ) -> Result<(), Error> {
+        for c in self.clone().into_bytes() {
```

Comment:
Replace the `for` loop with a call to [`util::write_items`](https://docs.rs/protocol/3.1.5/protocol/util/fn.write_items.html), which does the same thing. You can pass the `into_bytes()` struct into `write_items`